department-of-veterans-affairs/caseflow
# CASEFLOW-808 Add more actions during CAVC-Remand-Processed-Letter Response Window task

Resolves [CASEFLOW-808](https://vajira.max.gov/browse/CASEFLOW-808)

As a CAVC Litigation Support user, I want the ability to perform actions while the CAVC-Remand-Processed-Letter Response Window Task is on hold, so that I can act on the response from the Veteran before the 90 day window is complete.
If I end the hold, I can put the task back on hold for the completion of that window.

### Description
Add more actions to the CAVC-Remand-Processed-Letter Response Window task and adjust when the actions are available for different scenarios.

### Acceptance Criteria
- [x] Given the user is apart of the CAVC Litigation Support team, when they view the Case Details page of their assigned CAVC-Remand-Processed-Letter Response Window Task and click the Actions dropdown, then the end hold early as well as all available actions when ending the hold early is available
- [x] Verify the On-Hold Window task continues to display in the Org Assigned Tab
- [x] User cannot mark task complete without ending the hold
- [x] User would not have to end window task hold and do not have to recalculate # of days left. Ensure current org tabs remain the same.

Updated AC, explained in this [thread](https://dsva.slack.com/archives/CJL810329/p1612966041232900):
- [x] "Mark task complete" and "Extension request" should not be available unless the window task is first assigned to a user

### Testing Plan
Test the following for both CAVC admin and non-admin users.

1. Login as a CAVC lit support user
2. Go to http://localhost:3000/organizations/cavc-lit-support?tab=unassignedTab
3. Select an appeal with "Send CAVC-Remand-Processed Letter Task" to view the Case Details page. Note the docket number and UUID so you can find this appeal later.
4. Select "Mark as complete" and submit
5. Go to the team/org view; click the "Assigned" tab to find the appeal; click the appeal
6. Note the 5 actions to assign to other teams are available while the task is on hold
7. Select and submit "End hold early"
8. Note the 5 actions to assign to other teams are STILL available while the task is NOT on hold.
9. Note "Mark task complete" is now available, i.e., when the task is NOT on hold. (We'll test this near the end.)
10. Play around assigning tasks to other teams with and without the window task being on hold. Check the task tree and the appeals location in the correct queue tab. Verify the On-Hold Window task continues to display in the Org Assigned Tab
11. Try "Assign to person" and check the task tree. Note that the timed hold is not migrated/copied to the user task (bug ticket #15893) -- workaround: manually put the task on hold. Play around assigning tasks to other teams with and without the window task being on hold. Check the task tree and the appeals location in the correct queue tab. Verify the On-Hold Window task continues to display in the Org Assigned Tab. See screenshots below for the expected task actions when a task is assigned to person.
12. End the hold and click "Mark task complete" and submit.

### User Facing Changes

 BEFORE|AFTER
 ---|---
(on hold; time window is open) ![image](https://user-images.githubusercontent.com/55255674/107291778-4d214d00-6a2e-11eb-8043-a96e41ae651f.png) | (when not assigned to user) ![image](https://user-images.githubusercontent.com/55255674/107525146-2709c300-6b7c-11eb-8fe2-eb4358e7d0cc.png)<br/> (when assigned to user) ![image](https://user-images.githubusercontent.com/55255674/107557528-5c73d800-6b9f-11eb-8a47-ef42f48b316d.png)
(not on hold; time window ended) ![image](https://user-images.githubusercontent.com/55255674/107291135-4a722800-6a2d-11eb-89b9-4fa482fb6c91.png) | (when not assigned to user) ![image](https://user-images.githubusercontent.com/55255674/107527266-49044500-6b7e-11eb-88b6-55fd0b4aca16.png) <br/> (when assigned to user) ![image](https://user-images.githubusercontent.com/55255674/107525311-57e9f800-6b7c-11eb-9101-bf31a2268ba5.png)



### Code Documentation Updates
